File: 0429-n-ary-tree-level-order-traversal/0429-n-ary-tree-level-order-traversal.js
/**
 * // Definition for a Node.
 * function Node(val,children) {
 *    this.val = val;
 *    this.children = children;
 * };
 */

/**
 * @param {Node|null} root
 * @return {number[][]}
 */
var levelOrder = function(root) {
    const q = [root], res = []
    if (!root) return res
    while (q.length) {
        const len = q.length, lvl = []
        for (let i=0; i < len; i++) {
            const {val, children} = q.shift()
            lvl.push(val)
            for (const child of children) {
                q.push(child)
            }
        }
        res.push(lvl)
    }
    return res
};

File: 0429-n-ary-tree-level-order-traversal/0429-n-ary-tree-level-order-traversal.py
"""
# Definition for a Node.
class Node:
    def __init__(self, val=None, children=None):
        self.val = val
        self.children = children
"""

class Solution:
    def levelOrder(self, root: 'Node') -> List[List[int]]:
        if not root:
            return []
        q, res = [root], []
        while q:
            new_q, cur_lvl = deque(), []
            for node in q:
                cur_lvl.append(node.val)
                for child in node.children:
                    new_q.append(child)
            res.append(cur_lvl)
            q = new_q
        return res